lib/matplotlib/dates.py

```diff
@@ -1015,9 +1015,35 @@ def tick_values(self, vmin, vmax):
     def nonsingular(self, vmin, vmax):
         # whatever is thrown at us, we can scale the unit.
         # But default nonsingular date plots at an ~4 year period.
+        
         if vmin == vmax:
-            vmin = vmin - DAYS_PER_YEAR * 2
-            vmax = vmax + DAYS_PER_YEAR * 2
+            expander_factor = 0
+            dt = num2date(vmin)
+
+            # Order of Magnitude
+            if self.hms0d['bysecond'] != 0:
+                expander_factor = SEC_PER_MIN
+            elif self.hms0d['byminute'] != 0:
+                expander_factor = MIN_PER_HOUR
+            elif self.hms0d['byhour'] != 0:
+                expander_factor = HOURS_PER_DAY
+            elif dt.day > 1:
```

Comment:
I believe that is intended. If the view's min and max are equal, it is assumed that we are dealing with a singular point, and if so, we scale accordingly.